stake-pool/program/src/processor.rs

```diff
@@ -40,57 +46,52 @@ impl Processor {
         )
         .or(Err(Error::InvalidProgramAddress))
     }
+    /// Generates seed bump for stake pool authorities
+    pub fn find_authority_bump_seed(
+        program_id: &Pubkey,
+        my_info: &Pubkey,
+        authority_type: &[u8],
+    ) -> u8 {
+        let (_pubkey, bump_seed) =
+            Pubkey::find_program_address(&[&my_info.to_bytes()[..32], authority_type], program_id);
+        bump_seed
+    }
 
     /// Issue a stake_split instruction.
     pub fn stake_split<'a>(
-        stake_pool: &Pubkey,
         stake_account: AccountInfo<'a>,
         authority: AccountInfo<'a>,
-        nonce: u8,
         amount: u64,
         split_stake: AccountInfo<'a>,
+        signers: &[&[&[u8]]],
```

Comment:
Note that this is a problem with signers in general, but the `signers` parameter for these functions is confusing -- it's unclear how to construct it.  Instead of passing signers in here, could you pass all of the pubkeys and bump seed required and then construct it in this function?  That way the caller doesn't worry about how to create `signers`

Reply:
Previously it was exactly like you describe, however with new params for constructing program addresses (AUTHORITY_DEPOSIT or AUTHORITY_WITHDRAW) number of params in mint and burn functions reached 8. And cargo clippy did not like that (7 params per function maximum), this is why I refactored all signers construction code to happen outside of these functions.

Also there is a bonus now: we need to construct signers only once and then pass to each function when we chain several requests in one smart contract method.

If you think this is confusing, please tell me if you have any ideas on how to improve the situation without moving signers construction outside of those functions. Thanks.

Reply:
Ah gotcha, I see why you did it that way.  I think it's ok to reach 8 parameters in this case, so feel free to use `#[allow(clippy::too_many_arguments)]` before the functions.  To me, it'll make the interface simpler to use despite the higher number of arguments.